### PR TITLE
Fix deprecations and tests

### PR DESCRIPTION
**Pull request by gmalette**
This PR fixes deprecation warnings for Rails 5

### CODE REVIEW

**Comment by gmalette** [comment]
@nwtn @qq99

File: test/example/app/controllers/application_controller.rb [hunk 1]
```diff
@@ -1,8 +1,6 @@
 class ApplicationController < ActionController::Base
   protect_from_forgery
 
-  skip_before_filter :authenticate, :only => :unauthenticated
```
Comment (gmalette) [inline test/example/app/controllers/application_controller.rb]: I have no idea where this came from, but it was breaking because `:authenticate` doesn't exist

File: lib/turbograft.rb [hunk 2]
```diff
@@ -36,7 +36,7 @@ def referer
 
       ActiveSupport.on_load(:action_view) do
         (ActionView::RoutingUrlFor rescue ActionView::Helpers::UrlHelper).module_eval do
-          include XHRUrlFor
+          prepend XHRUrlFor
         end
       end unless RUBY_VERSION =~ /^1\.8/
```
Comment (gmalette) [inline lib/turbograft.rb]: Given we only test on Ruby 2.2, I was planning on bumping the gemspec to that and release `0.3.0`

**Comment by qq99** [comment]
cc @Shopify/tnt

**Comment by gmalette** [comment]
fyi @Shopify/scripts

**Comment by nwtn** [comment]
i don’t think i'll be very useful in reviewing this, so leaving it in the hands of the fine tnt folks

File: lib/turbograft.rb [hunk 3]
```diff
@@ -36,7 +36,7 @@ def referer
 
       ActiveSupport.on_load(:action_view) do
         (ActionView::RoutingUrlFor rescue ActionView::Helpers::UrlHelper).module_eval do
-          include XHRUrlFor
+          prepend XHRUrlFor
```
Comment (kurtfunai) [inline lib/turbograft.rb]: Just general curiosity here - why the change from include to prepend?

Comment (gmalette) [inline lib/turbograft.rb]: **TL, DR: ** `prepend` was added in Ruby 2, removing the need for `alias_method_chain`, which will be removed in Rails 5.

--- 
Say you want collections in Ruby  `module Identifiable` that gets included in other classes to extend the functionality the `id` method to also not return the name of the class.

```ruby
module Identifiable
  def id
    self.class_name + "." + super
  end
end

class User
  include Identifiable

  def id 
    primary_key
  end
end
```

## `include`

If you did this this way, Ruby will make `Identifiable` an ancestor of `User`. The ancestor chain of `User` would be 
```
[User, Identifiable, ...]
```

Notice `User` is first in the chain, thus calling `user.id` would simply return the primary key. Because of this, Rails implemented `alias_method_chain`, which you can use similarly to `alias_method_chain :id, :class_name`. 

This will do two things:
1 ) Rename the method `id` in `User` to `id_without_class_name`
2) Define a new `id` method like such:

```ruby
def id
  id_with_class_name
end
```

You would then define the `id_with_class_name` method in the inherited module (`Identifiable`) to do what you want, and instead of calling `super`, you would call `id_without_class_name`. This was basically a way to inject a more precise inheritance in the chain of the object when the language didn't support it. The dispatch diagram looks like something like this

```
+--------+          +----------------+
|        |          |                |
|  User  |          |  Identifiable  |
|        |          |                |
+--------+          +----------------+

  id  +------------->  id_with_class_name
                             +
                             |
  id_without_class_name <----+
  (originally called id)
```

## `prepend`

Using `prepend` flips things over. It prepends the module in the chain, so you get 

```ruby
[Identifiable, User, ...]
```

When you call `id`, it will hit the definition in `Identifiable` first, which calls `super`, which then finds the definition in `User`, resulting in the intended behaviour. 

The dispatch is much simpler to grok and debug

```
+---------------+   +--------+
|               |   |        |
| Identifiable  |   |  User  |
|               |   |        |
+---------------+   +--------+

  id   +----------->  id
```

Comment (kurtfunai) [inline lib/turbograft.rb]: Thank you for such a thorough and well written explanation 😃  🎆 

Makes a lot of sense.

Comment (qq99) [inline lib/turbograft.rb]: Really nice explanation, thanks!

**Comment by qq99** [comment]
Anyone available to 🎩 on admin?

**Comment by gmalette** [comment]
@Shopify/tnt halp

File: test/example/app/controllers/application_controller.rb (hunk 1 above)
Comment (gmalette) [inline test/example/app/controllers/application_controller.rb]: So this is because Rails 5 doesn't let you skip actions that are not applied.

Comment (gmalette) [inline test/example/app/controllers/application_controller.rb]: Must've been legacy and someone forgot to remove it some time ago

**Comment by justinthec** [comment]
I'll give it a :tophat: today

**Comment by justinthec** [comment]
🎩  💚 
Tested on OSX: Chrome, Firefox, and Safari

**Comment by gmalette** [comment]
🚢 ?

**Comment by qq99** [comment]
:shipit: imo

**Comment by gmalette** [comment]
fyi @rafaelfranca

File: lib/turbograft.rb [hunk 4]
```diff
@@ -36,7 +36,7 @@ def referer
 
       ActiveSupport.on_load(:action_view) do
         (ActionView::RoutingUrlFor rescue ActionView::Helpers::UrlHelper).module_eval do
```
Comment (phoet) [inline lib/turbograft.rb]: do you actually need to `module_eval` anything here? AFAIR `Module#include` and `Module#prepend` were made public in v2.1.

**Comment by justinthec** [comment]
image

```
Teaspoon running default suite at http://127.0.0.1:53984/teaspoon/default
rake aborted!
Errno::EINVAL: Invalid argument - bind(2) for "fe80::4001:aff:fef0:30" port 7055
```

@gmalette

File: lib/turbograft.rb (hunk 4 above)
Comment (rafaelfranca) [inline lib/turbograft.rb]: yeah, I think it is not needed.

**Comment by rafaelfranca** [comment]
Thanks!

**Comment by gmalette** [comment]
@justinthec is that related to my changes?

**Comment by justinthec** [comment]
Most likely, it only became red after your commit.

**Comment by gmalette** [comment]
Any idea what would cause that?
